### PR TITLE
Move some FoundationWall/Slab insulation properties into their Layer elements

### LINKED ISSUE #211: Move FoundationWall (and Slab) elements inside Insulation/Layer
If a `FoundationWall` has two continuous insulation layers (e.g., interior and exterior), there is currently no way to specify the `DistanceToTopOfInsulation` and `DistanceToBottomOfInsulation` properties for each of the layers.

To solve this, we should:
- `FoundationWall`: Move `DistanceToTopOfInsulation` and `DistanceToBottomOfInsulation` inside the `Insulation/Layer` element
- `Slab`: Move `PerimeterInsulationDepth` inside the `PerimeterInsulation/Layer` element
- `Slab`: Move `UnderSlabInsulationWidth` and `UnderSlabInsulationSpansEntireSlab` inside the `UnderSlabInsulation/Layer` element

This will take a little refactoring in order to have these elements only available in the `FoundationWall` (or `Slab`) `Insulation/Layer` elements and not _all_ surfaces.

### PR DESCRIPTION
**Pull request by yzhou601**
Closes #211.

FoundationWall previously:
```
          <FoundationWall>
            <DistanceToTopOfInsulation>0.0</DistanceToTopOfInsulation>
            <DistanceToBottomOfInsulation>2.0</DistanceToBottomOfInsulation>
            <Insulation>
              <Layer>
                <InstallationType>continuous - exterior</InstallationType>
                <NominalRValue>5.0</NominalRValue>
              </Layer>
              <Layer>
                <InstallationType>continuous - interior</InstallationType>
                <NominalRValue>10.0</NominalRValue>
              </Layer>
            </Insulation>
          </FoundationWall>
```

FoundationWall now:
```
          <FoundationWall>
            <Insulation>
              <Layer>
                <InstallationType>continuous - exterior</InstallationType>
                <NominalRValue>5.0</NominalRValue>
                <DistanceToTopOfInsulation>0.0</DistanceToTopOfInsulation>
                <DistanceToBottomOfInsulation>2.0</DistanceToBottomOfInsulation>
              </Layer>
              <Layer>
                <InstallationType>continuous - interior</InstallationType>
                <NominalRValue>10.0</NominalRValue>
                <DistanceToTopOfInsulation>0.0</DistanceToTopOfInsulation>
                <DistanceToBottomOfInsulation>8.0</DistanceToBottomOfInsulation>
              </Layer>
            </Insulation>
          </FoundationWall>
```

Slab previously:
```
          <Slab>
            <PerimeterInsulationDepth>2.0</PerimeterInsulationDepth>
            <UnderSlabInsulationWidth>4.0</UnderSlabInsulationWidth>
            <PerimeterInsulation>
              <Layer>
                <NominalRValue>5.0</NominalRValue>
              </Layer>
            </PerimeterInsulation>
            <UnderSlabInsulation>
              <Layer>
                <NominalRValue>5.0</NominalRValue>
              </Layer>
            </UnderSlabInsulation>
          </Slab>
```

Slab now:
```
          <Slab>
            <PerimeterInsulation>
              <Layer>
                <NominalRValue>5.0</NominalRValue>
                <InsulationDepth>2.0</InsulationDepth>
              </Layer>
            </PerimeterInsulation>
            <UnderSlabInsulation>
              <Layer>
                <NominalRValue>5.0</NominalRValue>
                <InsulationWidth>4.0</InsulationWidth>
              </Layer>
            </UnderSlabInsulation>
          </Slab>
```

### CODE REVIEW

File: schemas/BaseElements.xsd [hunk 1]
```diff
@@ -264,12 +264,12 @@
...
-					<xs:element name="DistanceToFoundationWallTop" type="LengthMeasurement" minOccurs="0">
+					<xs:element name="DistanceToTopOfInsulation" type="LengthMeasurement" minOccurs="0">
 						<xs:annotation>
 							<xs:documentation>[ft] Vertical distance from top of insulation layer to top of foundation wall.</xs:documentation>
 						</xs:annotation>
 					</xs:element>
-					<xs:element name="InsulationLayerHeight" type="LengthMeasurement" minOccurs="0">
+					<xs:element name="DistanceToBottomOfInsulation" type="LengthMeasurement" minOccurs="0">
 						<xs:annotation>
 							<xs:documentation>[ft] Height of vertical insulation layer.</xs:documentation>
```
Comment (shorowit) [inline schemas/BaseElements.xsd]: @yzhou601 Looks like this documentation needs to be updated too?

Comment (yzhou601) [reply schemas/BaseElements.xsd]: Yes I addressed it at the last commit

**Comment by shorowit** [comment]
HPXML WG was generally in favor of this change.

**Comment by shorowit** [comment]
@yzhou601 Can you bring this PR up to date?

**Comment by yzhou601** [comment]
@shorowit updated

**Pull request merged**